<?xml version="1.0" encoding="UTF-8" standalone="no" ?>
<launch>

	<!--arg name="ik_file" 	default="/srv/data/gait1992/inverse_kinematics/task_InverseKinematics.mot" />
	<arg name="model_file" 	default="/srv/data/gait1992/residual_reduction_algorithm/model_adjusted.osim" / -->
	<!-- arg name="ik_file" 	default="/catkin_ws/src/osrt_ros/data/Gait6_ik_clean.mot" /-->
	<arg name="ik_file" 	default="/catkin_ws/Data/test_ar_imu/UNKNOWN/0_ar_ik3_ik_upper.sto" />
	<arg name="model_file" 	default="/srv/data/mobl2016/mobl2016_v03.osim" />
	<arg name="rate_divider" default="1" />
	<arg name="async_run"    default="false"/>


        <node pkg="opensimrt_bridge" type="ik_dump" 	name="inverse_kinematics_from_file" output="screen">
		<param name="ik_file" 			value="$(arg ik_file)" />
                <param name="model_file" 		value="$(arg model_file)" />
                <param name="remove_n_last_rows" 	value="0" />
		<param name="simulation_loops" 		value="1" />
		
                <param name="resample_period" 		value="0.01" />
		<!-- maybe you want to run this node slower than the resampling period to give a lot of time between messages for the other algorithms to run. Otherwise this number should be one -->
		<param name="rate_divider" 		value="$(arg rate_divider)" />

		<remap from="/inverse_kinematics_from_file/output" to="/ik/output"/>
		<remap from="/inverse_kinematics_from_file/output_filtered" to="/ik/output_filtered"/>
		<remap from="/inverse_kinematics_from_file/out_labels" to="/ik/outlabels"/>
		
		<param name="async_run" 			value="$(arg async_run)"/>

		<!-- Filter values for filtered output -->
		<param name="filter_output" 		value="true" />
		<param name="memory" 			value="35" />
		<param name="cutoff_freq" 		value="6" />
		<param name="delay" 			value="14" />
		<param name="spline_order" 		value="3" />
	</node>

        <node pkg="osrt_ros" type="vis_ik" 	name="ik_visualization" output="screen">
                <param name="model_file" 		value="$(arg model_file)" />
		<param name="geometry_path" 		value="/srv/data/geometry_mobl" />
		<param name="which_model_1_2" 		value="1" />

		<remap from="/ik_visualization/input_filtered" to="/ik/output_filtered"/>
		<remap from="/in_labels" to="/ik/outlabels"/>
		
        </node>


</launch>
